lib/queue.js

```diff
@@ -1,256 +1,527 @@
-var redis = require('redis');
-var events = require('events');
-var util = require('util');
+'use strict';
 
-var Job = require('./job');
-var defaults = require('./defaults');
-var lua = require('./lua');
-var helpers = require('./helpers');
-var barrier = helpers.barrier;
+const redis = require('./redis');
+const Emitter = require('events').EventEmitter;
+const util = require('util');
+
+const Job = require('./job');
+const defaults = require('./defaults');
+const lua = require('./lua');
+const helpers = require('./helpers');
+const backoff = require('./backoff');
+const EagerTimer = require('./eager-timer');
 
 function Queue(name, settings) {
   if (!(this instanceof Queue)) {
     return new Queue(name, settings);
   }
 
+  Emitter.call(this);
```

Comment:
When I still naïvely expected this would be a small PR, I decided against switching to `class`es because it would increase the diff size. I like `class`es. I will do this. :)